tools/c7n_slacker/c7n_slacker/sqs_handler.py

```diff
@@ -0,0 +1,194 @@
+
+import boto3
+from botocore.exceptions import ClientError
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from ldap3 import Connection, Server
+from ldap3.core.exceptions import LDAPSocketOpenError
+
+from c7n import sqsexec, utils
+
+from slacker import NotifyMechanism
+
+
+class SQSHandler(object):
+
+    def __init__(self, config, logger):
+        self.config = config
+        self.session = self.session_factory(self.config)
+        self.logger = logger
+        self.base_dn = self.config.get('ldap_bind_dn')
+        self.email_key = self.config.get('ldap_email_key', 'mail')
+        self.uid_key = self.config.get('ldap_uid_attribute_name', 'sAMAccountName')
+        self.manager_attr = self.config.get('ldap_manager_attribute', 'manager')
+        self.attributes = ['displayName', self.uid_key, self.email_key, self.manager_attr]
+
+    def message_handler(self, connection, config, msg):
+        message = msg['Body']
+        try:
+            msg_json = json.loads(zlib.decompress(base64.b64decode(message)))
+            self.logger.info(
+                "Acct: %s,  msg: %s, resource type: %s, count: %d, policy: %s, \
+                recipients: %s, action_desc: %s, violation_desc: %s" % (
+                    msg_json.get('account', 'na'),
+                    msg['MessageId'],
+                    msg_json['policy']['resource'],
+                    len(msg_json['resources']),
+                    msg_json['policy']['name'],
+                    ', '.join(msg_json['action']['to']),
+                    msg_json['action']['action_desc'],
+                    msg_json['action']['violation_desc']))
+            self.logger.debug('Valid JSON')
+        except ValueError:
+            self.logger.warning("Invalid JSON")
+            return
+
+        if 'resource-owner' not in msg_json['action']['to']:
+            self.logger.debug("Resource owner indicator not found. Skipping message.")
+            return
+
+        resource_dict = {'account': msg_json['account'], 'account_id': msg_json['account_id'],
+                         'region': msg_json['region'], 'action_desc': msg_json['action']['action_desc'],
+                         'violation_desc': msg_json['action']['violation_desc']}
+
+        for resource in msg_json['resources']:
+            try:
+                resource_owner_value, matched_tag = self.get_resource_owner_values(resource)
+            except Exception as e:
+                self.logger.debug("Error fetching resource owner value: %s" % e)
+                continue
+
+            if resource_owner_value is None:
+                self.logger.debug("Resource details not found. Skipping message....")
+                continue
+            else:
+                resource_string = utils.resource_format(resource, msg_json['policy']['resource'])
+                self.logger.debug("resource string: %s", resource_string)
+
+                if (self.target_is_email(resource_owner_value)):
```

Comment:
This should be a separate function to return either `None` or `email`, shouldn't be exposing the logic on grabbing the email here as it's not appropriate for this level of abstraction